py-utils/src/utils/kv_store/kv_store_collection.py

```
@@ -280,10 +309,64 @@ def get(self, key):
                                         f"Key not present")
         return res
 
-    def set(self, key, value):
-        # TODO: Implement
-        pass
 
-    def delete(self, key):
-        # TODO: Implement
-        pass
+class PillarKvStore(KvStore):
+    """ Salt Pillar based KV Store """
+    name = "pillar"
+
+    def __init__(self, store_loc, store_path, delim='>'):
+        KvStore.__init__(self, store_loc, store_path, delim)
+        self._target = '*'
+    
+    def set_target(self, target:str='*') -> None:
+        """
+        tragents are required to run a slat command.
+        arguments:
+        target - target could be individual node name or * (for all minion)
+        """
+        ava_target = PillarKvStore.get_available_targets()
+        if target != '*' and target not in ava_target:
+            raise KvError(errno.ENOENT, "Target node not present in salt " \
```

It is perfectly ok for target to be new. Imagine this

kv = PillarKvStore(.../target). === Target may not exist. 
kv.set(k, v) === Values are stored in target passed in constructor as new key values.

---

now allowing any target to be passed. but check is done only at load method, Since in some cases wrong target raise silent errors.